### PR TITLE
controller, label, service: rm namespacelabeler

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/11480

It's not necessary anymore because we can create more flexible selector.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: service/controller/rbac/rbac.go [hunk 1]
```diff
@@ -44,11 +43,9 @@ func NewRBAC(config RBACConfig) (*RBAC, error) {
...
 	var namespaceAuthController *controller.Controller
 	{
-
-		namespaceSelector, err := labels.Parse(label.Organization)
-		if err != nil {
-			return nil, microerror.Mask(err)
-		}
+		selector := controller.NewSelector(func(labels controller.Labels) bool {
+			return labels.Has(label.Organization) || labels.Has(label.LegacyCustomer)
+		})
```
Comment (kopiczko) [inline service/controller/rbac/rbac.go:48]: We can do this now so we don't need the other controller.

File: pkg/label/label.go [hunk 2]
```diff
@@ -1,12 +1,10 @@
 package label
 
 const (
-	Cluster      = "giantswarm.io/cluster"
```
Comment (kopiczko) [inline pkg/label/label.go:4]: Do we need cluster label for anything?

**Comment by corest** [comment]
Tested, works fine

**Comment by kopiczko** [comment]
@corest tested it <3

**Pull request merged**